src/MediatR/Mediator.cs

```diff
@@ -226,6 +256,25 @@ private IEnumerable<TWrapper> GetNotificationHandlers<TWrapper>(object notificat
                 .ToList();
         }
 
+        private Task<TResponse> GetPipeline<TResponse>(object request, RequestHandlerDelegate<TResponse> invokeHandler)
+        {
+            var requestType = request.GetType();
+            var method = CreatePipelineMethod.MakeGenericMethod(requestType, typeof(TResponse));

```

Comment:
The problem with a compiled expression-based pipeline is that you lose the ability to inject services into your decorators/handlers' ctors. You have to know, at compile-time, what arguments the pipeline takes. That's why you typically end up with a `PipelineContext`, which essentially is a bag of objects, like the OWIN dictionary, that's used to flow stuff through the pipeline.

My question is; are we willing to drop the nice dependency injection in decorators/handlers for this? Has anyone done any benchmarking to see if this is needed? I don't think there's been a heap of complaints on MediatR perf to date. Has this changed?